pypeit/core/extract.py

```diff
@@ -2300,5 +2303,11 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
         canvas.add('constructedcanvas', canvas_list)
     # TODO two things need to be debugged. 1) For objects which were found and traced, i don't think we should be updating the tracing with
     # the PCA. This just adds a failutre mode. 2) The PCA fit is going wild for X-shooter. Debug that.
+    
```

Comment:
I suspect that a similar check needs to be added at the end of ech_objfind.py

Reply:
That is what this is! :)